sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -388,7 +388,7 @@ case class FileSourceScanExec(
     logInfo(s"Planning with ${bucketSpec.numBuckets} buckets")
     val filesGroupedToBuckets =
       selectedPartitions.flatMap { p =>
-        p.files.map { f =>
+        p.files.filter(_.getLen > 0).map { f =>
```

Comment:
do the filtering inside the map?

Reply:
Do we have a test case for this line?

Reply:
do you mean changing `filter...map...` to `flatMap`? I don't have a strong preference about it.

The updated test cases and the new test case are for this change.

Reply:
I personally prefer filter + map as it's shorter and clearer. I don't know if one is faster; two transformations vs having to return Some/None. For a Dataset operation I'd favor one operation, but this is just local Scala code.

Reply:
This `createBucketedReadRDD ` is for the bucket table, right?